app/components/StopInstancePrompt.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { type ReactNode } from 'react'
+
+import { apiQueryClient, useApiMutation, type Instance } from '@oxide/api'
+
+import { HL } from '~/components/HL'
+import { addToast } from '~/stores/toast'
+import { Button } from '~/ui/lib/Button'
+import { Message } from '~/ui/lib/Message'
+
+type StopInstancePromptProps = {
+  instance: Instance
+  children: ReactNode
+}
+
+export function StopInstancePrompt({ instance, children }: StopInstancePromptProps) {
+  const isStoppingInstance = instance.runState === 'stopping'
+
+  const stopInstance = useApiMutation('instanceStop', {
+    onSuccess: () => {
+      // trigger polling by the top level InstancePage one
```

Comment:
Oh this is very clever

Reply:
The only downside is that you have a delay in the loading state because it doesn't change to loading while the initial stop request goes through. I tried this

```ts
const isStoppingInstance = stopInstance.isPending || instance.runState === 'stopping'
```

and it does respond immediately, but it flashes back to off briefly in between the stop request completing and the invalidation running and flipping the instance to 'stopping'. I can't think of a fix other than adding back in the explicit state thing you had. Those things make me nervous because the logic has to be bulletproof to avoid accidentally getting stuck in a loading state, but I'm sure we can do it.